sherlock/internal/deprecated_models/v2models/changeset.go

```diff
@@ -254,18 +256,29 @@ func (s *internalChangesetStore) apply(db *gorm.DB, changesets []Changeset, user
 					fmt.Sprintf(config.Config.MustString("beehive.chartReleaseUrlFormatString"), chartRelease.Name),
 				)
 			}
-		}
 
-		for environmentID, chartReleaseNames := range environmentReleases {
-			environment, err := InternalEnvironmentStore.Get(db, Environment{Model: gorm.Model{ID: environmentID}})
-			if err == nil && environment.PagerdutyIntegration != nil && environment.PagerdutyIntegration.Key != nil {
-				go pagerduty.SendChangeSwallowErrors(
-					*environment.PagerdutyIntegration.Key,
-					fmt.Sprintf("Version changes to %s via Sherlock/Beehive", strings.Join(chartReleaseNames, ", ")),
-					fmt.Sprintf(config.Config.MustString("beehive.environmentUrlFormatString"), environment.Name),
+			for environmentID, chartReleaseNames := range environmentReleases {
+				environment, err := InternalEnvironmentStore.Get(db, Environment{Model: gorm.Model{ID: environmentID}})
+				if err == nil && environment.PagerdutyIntegration != nil && environment.PagerdutyIntegration.Key != nil {
+					go pagerduty.SendChangeSwallowErrors(
+						*environment.PagerdutyIntegration.Key,
+						fmt.Sprintf("Version changes to %s via Sherlock/Beehive", strings.Join(chartReleaseNames, ", ")),
+						fmt.Sprintf(config.Config.MustString("beehive.environmentUrlFormatString"), environment.Name),
+					)
+				}
+			}
```

Comment:
It looks like your edits accidentally nested this `range environmentReleases` loop inside the `range affectedChartReleases` loop. It should go after the `range affectedChartReleases`, not inside it.

Reply:
Can [affectedChartReleases](https://github.com/broadinstitute/sherlock/blob/27b38bac7dd272e00d90dffed52837a1cbd81ea5/sherlock/internal/deprecated_models/v2models/changeset.go#L167C2-L167C23) have different charts = different services? If so, I want to record each to pact.

Reply:
Affected chart releases is a list of each chart release that underwent a version change. I know you want to record each of those version changes to pact, that's totally fine. 

Here, it seems like you're modifying an unrelated part, the "per each environment report to pagerduty" code. I think the way you're changing it will make it not work so I think it would be better to leave it as-is.

Reply:
fe8c12d

Reply:
fe8c12d

Reply:
I'm not sure I understand this. Outer loop iterates through each affected chart release. For each iteration, I report to pact is an isolated if statement outside of the ```for environmentID, chartReleaseNames := range environmentReleases`` loop.

Reply:
a16498a

Reply:
Sorry I didn't see your comment -- your commit doesn't address the concern I'm raising. Can you see how your PR indents the `for environmentID, chartReleaseNames := range environmentReleases {` loop? It moves it inside the `for _, chartRelease := range affectedChartReleases {` loop when it wasn't there before.

![Screenshot 2023-08-14 at 11 12 10 AM](https://github.com/broadinstitute/sherlock/assets/29168264/1562324a-1707-401b-b733-f64823b3e10b)

Reply:
I got it : 774c032